File: src/main/java/org/ffpy/plugin/coding/service/SettingService.java
package org.ffpy.plugin.coding.service;

import com.sun.istack.Nullable;
import org.ffpy.plugin.coding.constant.TemplateName;

public interface SettingService {

    String getPackageName();

    void setPackageName(@Nullable String packageName);

    String getTranslateAppId();

    void setTranslateAppId(String appId);

    String getTranslateSecret();

    void setTranslateSecret(String secret);

    String getTemplate(TemplateName name);

    void setTemplate(TemplateName name, @Nullable String content);

    void reset();
}

File: src/main/kotlin/org/ffpy/plugin/coding/service/SettingService.kt
package org.ffpy.plugin.coding.service

import com.sun.istack.Nullable
import org.ffpy.plugin.coding.constant.TemplateName

interface SettingService {
    var translateAppId: String?
    var translateSecret: String?
    fun getTemplate(name: TemplateName): String?
    fun setTemplate(name: TemplateName, @Nullable content: String?)
    fun reset()
}